android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -259,7 +259,7 @@ public void setHardwareAccelerationDisabled(WebView view, boolean disabled) {
     if (disabled) {
```

Comment:
It looks like fix is correct, but need to determine if hardware rendering is available, see https://developer.android.com/guide/topics/graphics/hardware-accel#determining. 

```java
if (disabled || !view.isHardwareAccelerated())
```

Reply:
This is interesting. :)

Let me summarize the possible scenarios:
- _LAYER_TYPE_SOFTWARE_ -> will be fine on most situations, but people find that iframe videos are not playing
- _LAYER_TYPE_NONE_ -> people find that app may crash when a specific set of conditions are met (ie, Android 9? + using some kind of navigation libraries (with visual effect) + using WebView on the screen)
- _LAYER_TYPE_HARDWARE_ -> this would be new (in context of this library) and I agree that we will need to test more

From the [doc](https://developer.android.com/guide/topics/graphics/hardware-accel#view-level), it is like the existing code is already correct for "disabled" case by simply using `myView.setLayerType(View.LAYER_TYPE_SOFTWARE, null)`

Maybe we can make use of _isHardwareAccelerated()_ for the "else" case...
How about below code then?

```
} else {
  if (view.isHardwareAccelerated()) {
    view.setLayerType(View.LAYER_TYPE_HARDWARE, null);
  } else {
    view.setLayerType(View.LAYER_TYPE_NONE, null);
  }
}
```

Reply:
The critical point to decide is on the else=>else case. What should we set when we want to use the hardware (GPU) but it is not available. If it is supposed to fallback to use software, then the code will be what you've suggested earlier.

```
if (disabled || !view.isHardwareAccelerated()) {
  view.setLayerType(View.LAYER_TYPE_SOFTWARE, null);
} else {
  view.setLayerType(View.LAYER_TYPE_HARDWARE, null);
}
```

Reply:
how about

```java
if (disabled) {
  view.setLayerType(View.LAYER_TYPE_SOFTWARE, null);
} else {
  view.setLayerType(view.isHardwareAccelerated() ? View.LAYER_TYPE_HARDWARE : View.LAYER_TYPE_NONE, null);
}
```

Reply:
I like it but we have to decide whether we want to get rid of _LAYER_TYPE_NONE_ totally. 

If yes, then

```
if (disabled || !view.isHardwareAccelerated()) {
  view.setLayerType(View.LAYER_TYPE_SOFTWARE, null);
} else {
  view.setLayerType(View.LAYER_TYPE_HARDWARE, null);
}
```

If no, then

```
if (disabled) {
  view.setLayerType(View.LAYER_TYPE_SOFTWARE, null);
} else {
  view.setLayerType(view.isHardwareAccelerated() ? View.LAYER_TYPE_HARDWARE : View.LAYER_TYPE_NONE, null);
}
```

Personally I am convinced on the yes version.

Reply:
IIRC, NONE is causing issues, so it might be better to get rid of it.